gammapy/datasets/map.py

```diff
@@ -2506,16 +2508,41 @@ def update(self, exposure, psf, edisp, geom):
             Edisp map.
         geom : `WcsGeom`
             Counts geom
+        mask_safe_psf : `~gammapy.maps.Map`
+            Mask safe map of boolean type.
         """
         # TODO: simplify and clean up
         log.debug("Updating model evaluator")
         # cache current position of the model component
 
+        if mask_safe_psf is None:
+            mask = exposure.copy()
+            mask.data = (exposure.data != 0) & np.isfinite(exposure.data)
```

Comment:
I think we can safely assume that the `mask_safe` is "in sync" between the exposure and the IRF map (if this wasn't the case it would be a big problem anyway...). So creating the mask for the exposure should not yield any new information...